src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -164,16 +183,32 @@ public function save(CacheItemInterface $item)
         $value = $item["\0*\0value"];
         $expiry = $item["\0*\0expiry"];
 
-        if (null !== $expiry && $expiry <= microtime(true)) {
+        $now = microtime(true);
+
+        if (null !== $expiry && $expiry <= $now) {
             $this->deleteItem($key);
 
             return true;
         }
         if ($this->storeSerialized && null === $value = $this->freeze($value, $key)) {
             return false;
         }
-        if (null === $expiry && 0 < $item["\0*\0defaultLifetime"]) {
-            $expiry = microtime(true) + $item["\0*\0defaultLifetime"];
+        if ($this->maxLifetime && (null === $expiry || $expiry > $now + $this->maxLifetime)) {
```

Comment:
Might want to check if defaultLifetime is smaller then maxLifetime here before we use that?

Reply:
good catch, fixed